.github/workflows/workflow.yaml

```diff
@@ -10,37 +10,39 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  flake8:
-    timeout-minutes: 40
+  pre-commit-checks:
+    name: Pre-commit Checks
+    timeout-minutes: 30
     runs-on: ubuntu-latest
-
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-
-      - name: Setup mamba
-        uses: ./.github/actions/mamba
-
-      - name: Flake8
-        run: |
-          micromamba run -n modyn flake8 --version
-          micromamba run -n modyn flake8 modyn --statistics
+      - name: Checkout branch
+        uses: actions/checkout@v4
+      - name: Run pre-commit-conda
+        uses: quantco/pre-commit-conda@v1
 
   mypy-typechecking:
-    timeout-minutes: 20
+    name: Mypy Type Checks
     runs-on: ubuntu-latest
-
+    timeout-minutes: 20
+    defaults:
+      run:
+        shell: bash -el {0}
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-
-      - name: Setup mamba
-        uses: ./.github/actions/mamba
-
-      - name: Mypy
+      - name: Checkout branch
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Conda env
+        uses: mamba-org/setup-micromamba@f8b8a1e23a26f60a44c853292711bacfd3eac822
+        with:
+          environment-file: environment.yml
+          cache-environment: true
+      - name: Install repository
         run: |
-          micromamba run -n modyn mypy --version
-          micromamba run -n modyn mypy modyn
+          python -m pip install --no-build-isolation --no-deps --disable-pip-version-check -e .
+          micromamba run -n modyn pip install -r dev-requirements.txt
+      - name: Run mypy
+        run: mypy .
```

Comment:
Similarily here, we used to hide the setup of micromamba in the mamba action, not in the task itself. Any reason for changing the mypy job?

Reply:
If it's just the micromamba setup and nothing else, the indirection through a local actions doesn't really benefit us.

Reply:
But I mean the local action takes care of caching, no? In case of no updates, we can restore the environment from cache, speeding up the CI - that is what the local action does

Reply:
we already did that with the `cache-environment` micromamba flag, just with a different (default) cache-key. Now we use the default cache environment everywhere.